test/utility/misc.c

```diff
@@ -323,13 +325,90 @@ testutil_create_backup_directory(const char *home)
     size_t len;
     char *cmd;
 
-    len = strlen(home) * 4 + strlen(HOME_BACKUP_INIT_CMD) + 1;
+    len = strlen(home) * 6 + strlen(HOME_BACKUP_INIT_CMD) + 1;
     cmd = dmalloc(len);
-    testutil_check(__wt_snprintf(cmd, len, HOME_BACKUP_INIT_CMD, home, home, home, home));
+    testutil_check(
+      __wt_snprintf(cmd, len, HOME_BACKUP_INIT_CMD, home, home, home, home, home, home));
     testutil_checkfmt(system(cmd), "%s", "backup directory creation failed");
     free(cmd);
 }
 
+/*
+ * testutil_verify_src_backup --
+ *     Verify a backup source home directory against a backup directory for changes to blocks that
+ *     are not marked as changed.
+ */
+void
+testutil_verify_src_backup(WT_CONNECTION *conn, const char *backup, const char *home)
+{
+    struct stat sb;
+    WT_CURSOR *cursor, *file_cursor;
+    WT_DECL_RET;
+    WT_SESSION *session;
+    uint64_t cmp_size, offset, prev_offset, size, type;
+    int i, j, status;
+    char buf[1024], *filename, *id[WT_BLKINCR_MAX];
+    const char *idstr;
+
+    WT_CLEAR(buf);
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+    testutil_check(session->open_cursor(session, "backup:query_id", NULL, buf, &cursor));
+    i = 0;
+    while ((ret = cursor->next(cursor)) == 0) {
+        testutil_check(cursor->get_key(cursor, &idstr));
+        id[i++] = dstrdup(idstr);
+    }
+    testutil_check(cursor->close(cursor));
+    testutil_assert(i <= WT_BLKINCR_MAX);
+    /* Go through each id and open a backup cursor on it to test incremental values. */
+    for (j = 0; j < i; ++j) {
+        testutil_check(__wt_snprintf(buf, sizeof(buf), "incremental=(src_id=%s)", id[j]));
+        testutil_check(session->open_cursor(session, "backup:", NULL, buf, &cursor));
+        while ((ret = cursor->next(cursor)) == 0) {
+            testutil_check(cursor->get_key(cursor, &filename));
+            testutil_check(__wt_snprintf(buf, sizeof(buf), "incremental=(file=%s)", filename));
+            testutil_check(session->open_cursor(session, NULL, cursor, buf, &file_cursor));
+            prev_offset = 0;
+            while ((ret = file_cursor->next(file_cursor)) == 0) {
+                testutil_check(file_cursor->get_key(file_cursor, &offset, &size, &type));
+                /* We only want to check ranges for files. So if it is a full file copy, ignore. */
+                if (type != WT_BACKUP_RANGE)
+                    break;
+                testutil_check(__wt_snprintf(buf, sizeof(buf), "%s/%s", backup, filename));
+                ret = stat(buf, &sb);
+                /*
+                 * The file may not exist in the backup directory. If the stat call doesn't succeed
```

Comment:
No, it could also occur if a file was created after the last incremental backup. There are cases where a new file is returned (for backup) as a range for the whole file versus a file-copy. 

However, maybe it is worth changing the arg list to take an ID? Right now the function assumes it can work on either ID, (i.e. the copy is not specific to either). But if we wanted to only compare against one ID then we could do that.

Reply:
I had exactly that in https://github.com/wiredtiger/wiredtiger/pull/8840/commits/3ea0bd05520508975202b4db15bcd61001169a37